src/app/Processor/ProcessorEdit/useValidateConfigParams.ts

```diff
@@ -0,0 +1,98 @@
+import { useCallback, useState } from "react";
+import { omit } from "lodash";
+import { Action, Source } from "@openapi/generated";
+import { useTranslation } from "react-i18next";
+import { ConfigType, ProcessorValidation } from "../../../types/Processor";
+
+export function useValidateConfigParams(
+  config: Action | Source | undefined,
+  configTypes: ConfigType[]
+): {
+  validation: ProcessorValidation;
+  validate: () => boolean;
+  resetValidation: (propertyName: string) => void;
+  validateField: (typeName: string, propertyName: string) => boolean;
+} {
+  const [validation, setValidation] = useState<ProcessorValidation>({
+    isValid: undefined,
+    errors: {},
+  });
+  const { t } = useTranslation(["openbridgeTempDictionary"]);
+
+  const isEmpty = (value: string): boolean => {
+    return value.trim().length === 0;
+  };
+
+  const resetValidation = useCallback((propertyName: string): void => {
+    setValidation((prevState) => ({
+      ...prevState,
+      errors: omit(prevState.errors, propertyName),
+    }));
+  }, []);
+
+  const validateType = useCallback((): boolean => {
+    if (isEmpty(config?.type ?? "")) {
+      setValidation((prevState) => ({
+        ...prevState,
+        errors: { ...prevState.errors, type: t("common.required") },
+      }));
+      return false;
+    } else {
+      resetValidation("type");
+      return true;
+    }
+  }, [config, resetValidation, t]);
+
+  const validateField = useCallback(
+    (typeName: string, fieldName: string): boolean => {
+      const configType = configTypes.find((type) => type.value === typeName);
+      if (config && configType) {
+        const field = configType.fields.find(
+          (field) => field.name === fieldName
+        );
+        if (field) {
+          const fieldValidation = field.validate(config.parameters[fieldName]);
+          if (fieldValidation.isValid) {
+            resetValidation(field.name);
+          } else {
+            setValidation((prevState) => ({
+              ...prevState,
+              errors: {
+                ...prevState.errors,
+                [fieldName]: fieldValidation.errorMessage,
+              },
+            }));
+          }
+          return fieldValidation.isValid;
+        }
+      }
+      return false;
+    },
+    [config, configTypes, resetValidation]
+  );
+
+  const validate = useCallback((): boolean => {
```

Comment:
My aging brain would have appreciated some comments :rofl: 

e.g. "Collect separate validations for configured fields" etc.

Just an opinion, no need to change.

Reply:
Sorry, I've rushed a bit this PR. I've added some comments to explain what's happening inside `useValidateConfigParams`.
Let me know if I should do the same elsewhere.